pkg/controller/machine/controller.go

```diff
@@ -245,10 +250,15 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 
 	// Machine resource created. Machine does not yet exist.
 	klog.Infof("Reconciling machine object %v triggers idempotent create.", m.ObjectMeta.Name)
+	metrics.RegisterMachineCreate()
+	createStartTime := time.Now()
 	if err := r.actuator.Create(ctx, cluster, m); err != nil {
+		metrics.UpdateDurationFromStart(metrics.Create, createStartTime)
 		klog.Warningf("Failed to create machine %q: %v", name, err)
+		metrics.RegisterFailedMachineCreate(err.Error())
```

Comment:
Same here.  I think Create is failure-prone for async creation requests.

Reply:
Then, may be we can create a metric datapoint for failure if NodeRef is empty after a configurable duration of time?